### DISCUSSION TITLE
Advice as to how to compile inside of a monorepo

### DISCUSSION THREAD

**Discussion by pedropmedina** [jmcdo29/nest-commander#523] [Q&A]
Hi,

This might be a dump question, but could anyone show me how to compile a nest-commander app living inside of a nx monorepo. The nest-commander uses several of the services in the app as well as other libraries inside of the monorepo. I would like to compile with tsc or whatever the best options is in order to bring the nest-commander app outside and run without ts-node  as a dependency. 

Thank you.

**Comment by jmcdo29** [comment]
Generally speaking, I'd have two applications, one for the CLI one for the server, and each feature being made of use would be a library in the monorepo. Now, that's usually a lot of work to try to separate out, so a secondary option is to create a separate build config that uses the `cli.ts` as the `entryFile` instead of `main.ts` so that when `node dist/apps/app/main` is ran (or whatever the path is) then the CLI is called instead of the application. I plan to eventually get around to showing off the former kind of set up with an Nx monorepo

**Reply by pedropmedina** [reply]
Hi @jmcdo29,

Thanks for the quick response. I thought about building a new app just for the cli, but its purpose is so simple that it is not worth it. Also I have several services built into the main app that don't belong in any lib really and I would like to make use of in the cli. 

Anyway for those out there that need a solution to this,  Nx has "additionalEntryPoints" prop in "@nrwl/node:webpack" executor which makes it pretty trivial to achieve this. Call the build executor and that's it. You'll end up with main.js and cli.js with a setup as shown in the snippet below:

```json  
"build": {
      "executor": "@nrwl/node:webpack",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/apps/mainAppDir",
        "main": "apps/mainAppDir/src/main.ts",
        "additionalEntryPoints": [
          { "entryName": "cli", "entryPath": "apps/mainAppDir/src/commands/main.ts" }
        ],
        "tsConfig": "apps/mainAppDir/tsconfig.app.json",
        "assets": ["apps/mainAppDir/src/assets"]
      },
      "configurations": {
        "production": {
          "optimization": true,
          "extractLicenses": true,
          "inspect": false,
          "generatePackageJson": true,
          "fileReplacements": [
            {
              "replace": "apps/mainAppDiri/src/environments/environment.ts",
              "with": "apps/mainAppDir/src/environments/environment.prod.ts"
            }
          ]
        }
      }
    }
```

Thank you again!
